src/test/rustdoc/sidebar-items.rs

```diff
@@ -1,16 +1,23 @@
+#![feature(associated_type_defaults)]
```

Comment:
Adding this feature flag enables me to test all permutations of `{provided,required}.{associated-{consts,types},methods}` at the same place. I can split this test if you don't like to see a feature flag for this rather general test.